src/transform/src/dataflow.rs

```diff
@@ -28,7 +28,7 @@ use crate::{monotonic::MonotonicFlag, Optimizer, TransformError};
 /// information to dataflow sources and lifts monotonicity information.
 pub fn optimize_dataflow(
     dataflow: &mut DataflowDesc,
-    indexes: &HashMap<GlobalId, Vec<(GlobalId, Vec<MirScalarExpr>)>>,
+    indexes: &dyn IndexOracle,
```

Comment:
Does this not need to be boxed because it's a reference?

Reply:
Yep! Trait objects just need to be behind a pointer of some sort. Either a `Box` or a reference will do.

Reply:
Also `Arc<dyn Trait>` and `Rc<dyn Trait>`. Still waiting on `Vec<dyn Trait>` but sort of neat, imo.